Help Testcontainers on Jenkins. DSDEGP-1369

kshakir opened the pull request:
### Description

Our Jenkins setup required a couple tweaks to the Testcontainers specs, so that 1) ES didn't think it was running the unit tests in production, and 2) if the docker daemon prepends "docker.io/" to the image name, the Testcontainer still works.

DSDEGP-1369

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Move the JIRA issue to QA once this checklist is completed

pshapiro4broad commented on project/src/main/scala/org/broadinstitute/clio/integrationtest/ClioIntegrationTestPlugin.scala:
```diff
@@ -5,6 +5,7 @@ import sbt._
 /**
   * A plugin for creating an integration test task, that runs via docker.
   */
+//noinspection TypeAnnotation
```
Is there no better way to fix the compiler warnings here?

pshapiro4broad commented on src/test/scala/org/broadinstitute/clio/dataaccess/ElasticsearchContainer.scala:
```diff
@@ -1,28 +1,56 @@
+    // `new RemoteDockerImage(image).get()`. If the first attempt fails, it is logged by Testcontainers during this
+    // resolution. If/when the second attempt fails on the modified image name, the latter error is printed during
+    // `val container` below.
+    val image = TestContainers.DockerImages.elasticsearch
```
As a workaround, this seems OK to me. It wouldn't be that much harder to make a general version, though.

pshapiro4broad reviewed (approved):
👍 Although I'd prefer to modify the code rather than add directives to disable inspections. Maybe that can wait for a future checkin though,.

pshapiro4broad commented:
test this please

pshapiro4broad reviewed (approved):
👍 
Nice to see the type warnings fixed, I hope it wasn't too painful finding the correct types.

kshakir committed:
Fix (most) inspections in IntelliJ.
For those that can't be fixed, added a comment why.

kshakir committed:
Help Testconatainers on Jenkins.
When the regular image name fails, try again with a "docker.io/" prefix.
Convince Elasticsearch that it's running on localhost.
